test/end-to-end/test_launcher_restarts_supervisor.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -eou pipefail
+
+wait_for_sup_to_start() {
+	until pgrep hab-sup &>/dev/null; do
```

Comment:
It'd be good to have some kind of timeout here, in case something goes wrong.

Reply:
Since we can't actually run this as an automated test, I opted for the simpler route. I'm planning to get all the end-to-end test scripts running on Buildkite since they're not simple to get running in travis. Until that time, I'm trying to keep the scripts as simple as possible since it may make sense to [add logic like timeouts at a higher level](https://linux.die.net/man/1/timeout) rather than duplicating a bunch of logic in the test.

Reply:
Cool; add that as a comment and :shipit: